From c0a23ae4f08b1d85791cb1a81456746c82f62b0c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:11 -0400
Subject: [ethernet] i40e: notify VF of all types of resets

Message-id: <1409658042-22689-131-git-send-email-sassmann@redhat.com>
Patchwork-id: 90322
O-Subject: [RHEL7.1 PATCH 130/281] i40e: notify VF of all types of resets
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Currently, the PF driver only notifies the VFs for PF reset events.
Instead, notify the VFs for all types of resets, so they can attempt a
graceful reinit.

Change-ID: I03eb7afde25727198ef620f8b4e78bb667a11370
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 263fc48f97f603c348e44ed1f06f06320beed662)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 3115b5e..d7c5cc8 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4412,6 +4412,9 @@ void i40e_do_reset(struct i40e_pf *pf, u32 reset_flags)
 
  WARN_ON(in_interrupt());
 
+ if (i40e_check_asq_alive(&pf->hw))
+  i40e_vc_notify_reset(pf);
+
  /* do the biggest reset indicated */
  if (reset_flags & (1 << __I40E_GLOBAL_RESET_REQUESTED)) {
 
@@ -5328,9 +5331,6 @@ static int i40e_prep_for_reset(struct i40e_pf *pf)
 
  dev_dbg(&pf->pdev->dev, "Tearing down internal switch for reset\n");
 
- if (i40e_check_asq_alive(hw))
-  i40e_vc_notify_reset(pf);
-
  /* quiesce the VSIs and their queues that are not already DOWN */
  i40e_pf_quiesce_all_vsi(pf);
 
-- 
1.7.1